driver/razeraccessory_driver.c

```diff
@@ -155,7 +157,19 @@ static ssize_t razer_attr_read_test(struct device *dev, struct device_attribute
 static ssize_t razer_attr_write_mode_spectrum(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
 {
     struct razer_accessory_device *device = dev_get_drvdata(dev);
-    struct razer_report report = razer_chroma_standard_matrix_effect_spectrum(VARSTORE, BACKLIGHT_LED);
+    struct razer_report report = { 0 };
+
+    switch (device->usb_dev->descriptor.idProduct)
+    {
+    case USB_DEVICE_ID_RAZER_CHROMA_MUG:
+        report = razer_chroma_standard_matrix_effect_spectrum(VARSTORE, BACKLIGHT_LED);
+        break;
+
+    case USB_DEVICE_ID_RAZER_CHROMA_HDK:
+        report = razer_chroma_extended_matrix_effect_spectrum(VARSTORE, ZERO_LED);
+        break;
+    }
```

Comment:
Can you make either one of the two options a `default` or add a printk as error and a return?